### ISSUE TITLE
BUG: no video files

### ISSUE THREAD

**Issue by ZashIn** [tobexyz/yaacc-code#88]
Video files are not shared, even though "serve video is enabled" with a local server running.
Under content a "Video" folder shows up, but it is always empty.

Access of the video folder results in the following error:
```
10035 10035 E ResolveError: The following error occurred: UpnpFailure [invocation=(ActionInvocation) (Action, Arguments: 10) Browse, defaultMsg=Error: Current state of service prevents invoking that action. java.lang.NumberFormatException: Cannot parse null string.](400999)
```

Music and pictures show up correctly.

In contrast, video files from other UPNP / DLNA servers, like Kodi, show up correctly in the network (shared from the same device as with YAACC).

Tested with:
YAACC 4.0.2
Android 14
Pixel 4a

**Comment by tobexyz** [comment]
Hi, thank you for your report!
I can't confirm the bug on my device, but the NumberFormatException might be a problem :-)

Can you send me the stack trace shown in the log, please?

Did you try pushing the update media store button on the YACC server entry in the sender list?

Can you tell me the file type of one of your video files on your device you expected to shown up, please?

I have tested the video sharing using MP4 files mostly

**Comment by ZashIn** [comment]
With higher log level, the log viewer gets quite laggy, but I could extract this much:
```
12-30 18:06:30.149 23204 23294 D de.yaacc.upnp.server.contentdirectory.VideosFolderBrowser: VideoItem: 9934 Name: <NAME>.mp4 uri: http://<IP>:49157/res/9934/file.mp4
12-30 18:06:30.151 23204 23294 D de.yaacc.upnp.server.contentdirectory.YaaccContentDirectory: exception on browse
12-30 18:06:30.151 23204 23294 D de.yaacc.upnp.server.contentdirectory.YaaccContentDirectory: java.lang.NumberFormatException: Cannot parse null string
12-30 18:06:30.151 23204 23294 D de.yaacc.upnp.server.contentdirectory.YaaccContentDirectory: 	at java.lang.Long.parseLong(Long.java:714)
12-30 18:06:30.151 23204 23294 D de.yaacc.upnp.server.contentdirectory.YaaccContentDirectory: 	at java.lang.Long.parseLong(Long.java:876)
12-30 18:06:30.151 23204 23294 D de.yaacc.upnp.server.contentdirectory.YaaccContentDirectory: 	at de.yaacc.upnp.server.contentdirectory.YaaccContentDirectory.formatDuration(YaaccContentDirectory.java:464)
12-30 18:06:30.151 23204 23294 D de.yaacc.upnp.server.contentdirectory.YaaccContentDirectory: 	at de.yaacc.upnp.server.contentdirectory.VideosFolderBrowser.browseItem(VideosFolderBrowser.java:95)
12-30 18:06:30.151 23204 23294 D de.yaacc.upnp.server.contentdirectory.YaaccContentDirectory: 	at de.yaacc.upnp.server.contentdirectory.ContentBrowser.browseChildren(ContentBrowser.java:65)
12-30 18:06:30.151 23204 23294 D de.yaacc.upnp.server.contentdirectory.YaaccContentDirectory: 	at de.yaacc.upnp.server.contentdirectory.YaaccContentDirectory.browse(YaaccContentDirectory.java:387)
12-30 18:06:30.151 23204 23294 D de.yaacc.upnp.server.contentdirectory.YaaccContentDirectory: 	at de.yaacc.upnp.server.contentdirectory.YaaccContentDirectory.browse(YaaccContentDirectory.java:324)
12-30 18:06:30.151 23204 23294 D de.yaacc.upnp.server.contentdirectory.YaaccContentDirectory: 	at java.lang.reflect.Method.invoke(Native Method)
12-30 18:06:30.151 23204 23294 D de.yaacc.upnp.server.contentdirectory.YaaccContentDirectory: 	at org.seamless.util.Reflections.invoke(Reflections.java:41)
12-30 18:06:30.151 23204 23294 D de.yaacc.upnp.server.contentdirectory.YaaccContentDirectory: 	at org.fourthline.cling.model.action.MethodActionExecutor.execute(MethodActionExecutor.java:90)
12-30 18:06:30.151 23204 23294 D de.yaacc.upnp.server.contentdirectory.YaaccContentDirectory: 	at org.fourthline.cling.model.action.AbstractActionExecutor$1.execute(AbstractActionExecutor.java:73)
12-30 18:06:30.151 23204 23294 D de.yaacc.upnp.server.contentdirectory.YaaccContentDirectory: 	at org.fourthline.cling.model.DefaultServiceManager.execute(DefaultServiceManager.java:128)
12-30 18:06:30.151 23204 23294 D de.yaacc.upnp.server.contentdirectory.YaaccContentDirectory: 	at org.fourthline.cling.model.action.AbstractActionExecutor.execute(AbstractActionExecutor.java:71)
12-30 18:06:30.151 23204 23294 D de.yaacc.upnp.server.contentdirectory.YaaccContentDirectory: 	at org.fourthline.cling.controlpoint.ActionCallback.run(ActionCallback.java:123)
12-30 18:06:30.151 23204 23294 D de.yaacc.upnp.server.contentdirectory.YaaccContentDirectory: 	at org.fourthline.cling.support.contentdirectory.callback.Browse.run(Browse.java:89)
12-30 18:06:30.151 23204 23294 D de.yaacc.upnp.server.contentdirectory.YaaccContentDirectory: 	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:487)
12-30 18:06:30.151 23204 23294 D de.yaacc.upnp.server.contentdirectory.YaaccContentDirectory: 	at java.util.concurrent.FutureTask.run(FutureTask.java:264)
12-30 18:06:30.151 23204 23294 D de.yaacc.upnp.server.contentdirectory.YaaccContentDirectory: 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
12-30 18:06:30.151 23204 23294 D de.yaacc.upnp.server.contentdirectory.YaaccContentDirectory: 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:644)
12-30 18:06:30.151 23204 23294 D de.yaacc.upnp.server.contentdirectory.YaaccContentDirectory: 	at java.lang.Thread.run(Thread.java:1012)
12-30 18:06:30.151 23204 23294 I DefaultServiceManager: Releasing lock
12-30 18:06:30.151 23204 23204 D de.yaacc.browser.BrowseItemLoadTask: Ended AsyncTask for loading:de.yaacc.upnp.callback.contentdirectory.ContentDirectoryBrowseResult@ed56111
12-30 18:06:30.151 23204 23204 E ResolveError: The following error occurred: UpnpFailure [invocation=(ActionInvocation) (Action, Arguments: 10) Browse, defaultMsg=Error: Current state of service prevents invoking that action. java.lang.NumberFormatException: Cannot parse null string.](400999)
```
Video files are mostly either `.mp4` or `.mkv` with x265 HEVC / AAC codecs, either from camera or downloaded.

Updating the media store does not show video files either.

**Comment by tobexyz** [comment]
Hi, thanks to your report it was easy to find the root cause. The error occurs, if the duration in the media store is null. The code is now changed by stetting the duration to zero in this case. 

If you are able to compile the project, check it out that way. Otherwise you can install the attached debug apk, which has to be extracted first
[yaacc-debug.zip](https://github.com/tobexyz/yaacc-code/files/13801853/yaacc-debug.zip)

**Comment by ZashIn** [comment]
Thank you for the quick fix! Video content does show up now, both in YACC and on the UPNP device.
Although, I cannot test playback at the moment, since I have no TV at hand and I get currently nothing (music, pictures neither) streamed from YACC to PC. But that is probably a different issue, since it worked with a TV last week and applies to both old and new version.

Anyways tanks and happy new year!

**Comment by tobexyz** [comment]
just released the new version 4.0.3. Normally it will appear in a few days on f-droid...

Happy new year to you too!
